File: WeiBoNews/DataBaseManager.m
//
//  DataBaseManager.m
//  CreateDataBase
//
//  Created by  刘洪君 on 14-5-10.
//  Copyright (c) 2014年 LHJ. All rights reserved.
//

#import "DataBaseManager.h"

#define kDefaultDBName @"account.db"

@implementation DataBaseManager
static DataBaseManager * _sharedDBManager;

+ (DataBaseManager *) defaultDBManager {
	if (!_sharedDBManager) {
		_sharedDBManager = [[DataBaseManager alloc] init];
	}
	return _sharedDBManager;
}

- (id) init {
    self = [super init];
    if (self) {
        int state = [self initializeDBWithName:kDefaultDBName];
        if (state == -1) {
            NSLog(@"数据库初始化失败");
        } else {
            NSLog(@"数据库初始化成功");
        }
    }
    return self;
}

- (int) initializeDBWithName : (NSString *) name {
    if (!name) {
		return -1;  // 返回数据库创建失败
	}
    // 沙盒Docu目录
    NSString * docp = [NSSearchPathForDirectoriesInDomains(NSDocumentDirectory,NSUserDomainMask, YES) objectAtIndex:0];
	_name = [docp stringByAppendingString:[NSString stringWithFormat:@"/%@",name]];
    NSLog(@"数据库路径：%@", _name);
	NSFileManager * fileManager = [NSFileManager defaultManager];
    BOOL exist = [fileManager fileExistsAtPath:_name];
    [self connect];
    if (!exist) {
        return 0;
    } else {
        return 1;// 返回数据库已经存在
	}
}
/// 连接数据库

- (void) connect {
	if (!self.dataBase) {
		self.dataBase = [[FMDatabase alloc] initWithPath:_name];
	}
    self.dataBase.shouldCacheStatements = YES;
	if (![_dataBase open]) {
		NSLog(@"不能打开数据库");
	}
}













@end

File: WeiBoNews/DataManager/DataBaseManager.swift
//
//  DataBaseManager.swift
//  WeiBoNews
//
//  Created by 123456 on 16/11/22.
//  Copyright © 2016年 123456. All rights reserved.
//

import UIKit

class DataBaseManager: NSObject {
    var dataBase: FMDatabase?
    var docName: String?
    
    class var sharedInstance :DataBaseManager {
        struct Singleton {
            static let instance = DataBaseManager()
        }
        return Singleton.instance
    }
    
    override init() {
        super.init()
        initializeDBWithName(name: "account.db")
    }
    
    
    func initializeDBWithName(name: String) {
        if name.isEmpty {
            return
        }
        
//        guard (dataBase?.open())! else {
//            print("Unable to open database")
//            return
//        }
        
        let document = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
        
        docName = String.init(format: "%@/%@", document, name) 
        print("数据库路径：%@", docName)
        
        let fileManager = FileManager.default
        let exist = fileManager.fileExists(atPath: docName!)
        self.connect()
        if exist == false {
            print("数据库已经存在")
        }else {
            print("数据库不存在")
        }
    }
    
    func connect() {
        if dataBase == nil {
            dataBase = FMDatabase.init(path: docName)
        }
//        dataBase?.shouldCacheStatements() = true
//        dataBase?.open()
        if dataBase?.open() == false {
            print("不能打开数据库")
        }else {
            print("可以打开数据库")
        }

    }

}



//Optional("/Users/a123456/Library/Developer/CoreSimulator/Devices/FCE9F433-0084-4C23-980F-06E5F4F51258/data/Containers/Data/Application/CF105E2C-85FE-46F4-8FF1-2E2086B36C26/Documents/account.db")
